packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -0,0 +1,348 @@
+
+			$namespacesToScan[ $namespace ][] = array(
+				'paths'   => array_map(
+					function( $item ) {
+						return $item['path']; },
+					$info
+				),
+				'version' => $version,
+			);
+		}
+
+		krsort( $namespacesToScan );
+
+		foreach ( $namespacesToScan as $namespace => $groups ) {
+
+			foreach ( $groups as $group ) {
+
+				foreach ( $group['paths'] as $dir ) {
+					$dir = $filesystem->normalizePath( $filesystem->isAbsolutePath( $dir ) ? $dir : $basePath . '/' . $dir );
+
+					if ( ! is_dir( $dir ) ) {
+						continue;
+					}
+
+					$namespaceFilter = '' === $namespace ? null : $namespace;
+					$classMap        = $this->addClassMapCode(
+						$filesystem,
+						$basePath,
+						$vendorPath,
+						$dir,
+						$blacklist,
+						$namespaceFilter,
+						$group['version'],
+						$classMap
+					);
+				}
+			}
+		}
+
+		return $classMap;
+	}
+
+	/**
+	 * Add a single class map resolution.
+	 *
+	 * @param Filesystem $filesystem Filesystem class instance.
+	 * @param string     $basePath Base path.
+	 * @param string     $vendorPath Path to the vendor diretory.
+	 * @param string     $dir Direcotry path.
+	 * @param null       $blacklist Blacklist of namespaces set to be ignored currently not used.
+	 * @param null       $namespaceFilter Namespace being used.
+	 * @param string     $version The version of the package.
+	 * @param array      $classMap The current classMap.
+	 *
+	 * @return array
+	 */
+	private function addClassMapCode(
+		$filesystem,
+		$basePath,
+		$vendorPath,
+		$dir,
+		$blacklist = null,
+		$namespaceFilter = null,
+		$version,
+
+		array $classMap = array()
+	) {
+
+		foreach ( $this->generateClassMap( $dir, $blacklist, $namespaceFilter ) as $class => $path ) {
+			$pathCode = "array( 'path' => " . $this->getPathCode( $filesystem, $basePath, $vendorPath, $path ) . ", 'version'=>'" . $version . "' ),\n";
+
+			if ( ! isset( $classMap[ $class ] ) ) {
+				$classMap[ $class ] = $pathCode;
+			} elseif ( $this->io && $classMap[ $class ] !== $pathCode && ! preg_match(
+				'{/(test|fixture|example|stub)s?/}i',
+				strtr( $classMap[ $class ] . ' ' . $path, '\\', '/' )
+			)
+			) {
+				$this->io->writeError(
+					'<warning>Warning: Ambiguous class resolution, "' . $class . '"' .
+					' was found in both "' . str_replace(
+						array( '$vendorDir . \'', "',\n" ),
+						array( $vendorPath, '' ),
+						$classMap[ $class ]
+					) . '" and "' . $path . '", the first will be used.</warning>'
+				);
+			}
+		}
+
+		return $classMap;
+	}
+
+	/**
+	 * Trigger the class map generation.
+	 *
+	 * @param string $dir  Directory path.
+	 * @param null   $blacklist Blacklist of namespaces set to be ignored currently not used.
+	 * @param null   $namespaceFilter Namespace being used.
+	 * @param bool   $showAmbiguousWarning Whether to show a warning in the console.
+	 *
+	 * @return array
+	 */
+	private function generateClassMap( $dir, $blacklist = null, $namespaceFilter = null, $showAmbiguousWarning = true ) {
```

Comment:
We can't remove this one because it's private and we use it directly in this class, right?